src/app/clusters/energy-preference-server/energy-preference-server.cpp

```diff
@@ -0,0 +1,224 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "energy-preference-server.h"
+
+#include <app/util/attribute-storage.h> // Needed for registerAttributeAccessOverride
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/callback.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/ConcreteAttributePath.h>
+#include <lib/core/CHIPEncoding.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::EnergyPreference;
+using namespace chip::app::Clusters::EnergyPreference::Structs;
+using namespace chip::app::Clusters::EnergyPreference::Attributes;
+
+using Status = Protocols::InteractionModel::Status;
+
+namespace {
+
+class EnergyPrefAttrAccess : public AttributeAccessInterface
+{
+public:
+    EnergyPrefAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), EnergyPreference::Id) {}
+
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+};
+
+EnergyPrefAttrAccess gEnergyPrefAttrAccess;
+Delegate * gsDelegate = nullptr;
+
+CHIP_ERROR EnergyPrefAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
```

Comment:
It seems you only implement read part, and not implement set part..

Reply:
It is a Fixed attribute, there is no Set